CONTRIBUTING.md

```diff
@@ -0,0 +1,119 @@
+
+Writing documentation tests is needed for better illustration of the added/altered capabilities for end-users of the crate.
+
+#### Unit tests
+
+[Unit tests][2] don't have much sense when it comes to macro testing. That's why they are rare beasts in the code of this repository. However, occasionally, they're very useful for testing some complicated properties locally (like correctness of syntax parsing, for example).
+
+
+### Review
+
+To get merged, any [PR] should be reviewed and approved by at least one of the active project maintainers, except the [PR] submitter, of course.
+
+Furthermore, __any `derive_more` community member is welcome to review any [PR].__
+
+All `derive_more` contributors who choose to review and provide feedback on [Pull Requests][PR] have a responsibility to both the project and the individual making the contribution. Reviews and feedback must be helpful, insightful, and geared towards improving the contribution as opposed to simply blocking it. If there are reasons why you feel the [PR] should not land, explain what those are. Do not expect to be able to block a [Pull Request][PR] from advancing simply because you say "No" without giving an explanation. Be open to having your mind changed. Be open to working with the contributor to make the [Pull Request][PR] better.
+
+When reviewing a [Pull Request][PR], the primary goals are for the codebase to improve and for the person submitting the request to succeed. Even if a [Pull Request][PR] doesn't land, the submitters should come away from the experience feeling like their effort was not wasted or unappreciated.
+
+#### Abandoned or stalled [Pull Requests][PR]
+
+If a [Pull Request][PR] appears to be abandoned or stalled, it's polite to first check with the contributor to see if they intend to continue the work before checking if they would mind if you took it over (especially if it just has nits left). When doing so, it's courteous to give the original contributor credit for the work they started (by preserving their name and email address with `Co-authored-by:` meta-data tag in the commit).
+
+
+### Merging
+
+All [PR]s are [squash]-merged to preserve the main history linear and meaningful.
+
+#### Commit message for a [squash] merge
+
+Commit message for a [squash] merge of the [PR] should mention its number and the number of the relevant [issue] (if it has the one).
+
+
+
+
+## Releasing
+
+To produce a new release of the `derive_more` crate, perform the following steps:
+
+1. Bump up the crate version correctly in [`Cargo.toml`] manifest.
+
+2. Complete the existing [CHANGELOG] or fill up a new one for the new version declared in [`Cargo.toml`].
+
+3. Update [README] with the new version declared in [`Cargo.toml`].
+
+4. Apply the Git version tag (in `v<version>` format) and push it.
+
+5. Wait for the CI pipeline to complete successfully, and the [GitHub release] being created.
+
+6. Publish the crate on [crates.io] via `cargo publish` command.
```

Comment:
I use [`cargo release`](https://github.com/crate-ci/cargo-release) to publish which removes some of these steps. 
```suggestion
1. Complete the existing [CHANGELOG] or fill up a new one for the new version.
2. Update [README] installation instructions with the new version.
3. Run `cargo release patch` (or `minor` or `major`)
4. Wait for the CI pipeline to complete successfully, and the [GitHub release] being created.
```